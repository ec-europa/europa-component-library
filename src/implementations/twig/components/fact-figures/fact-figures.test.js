import {
  merge,
  renderTwigFileAsNode,
  renderTwigFileAsHtml,
} from '@ecl/test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';

// Import data for tests
import demoContent from '@ecl/specs-component-fact-figures/demo/data';

const demo1Col = { ...demoContent, column: 1 };
demo1Col.items = demo1Col.items.slice(0, 1);
const demo2Col = { ...demoContent, column: 2 };
demo2Col.items = demo2Col.items.slice(0, 2);
const demo3Col = { ...demoContent, column: 3 };
demo3Col.items = demo3Col.items.slice(0, 6);
const demo4Col = { ...demoContent, column: 4 };

expect.extend(toHaveNoViolations);

describe('Fact and figures', () => {
  const template = '@ecl/fact-figures/fact-figures.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('1 col', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(demo1Col)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(demo1Col, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(demo1Col, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });

    test(`passes the accessibility tests`, async () => {
      expect(
        await axe(renderTwigFileAsHtml(template, demo1Col, true)),
      ).toHaveNoViolations();
    });
  });

  describe('2 cols', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(demo2Col)).resolves.toMatchSnapshot();
    });
  });

  describe('3 cols', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(demo3Col)).resolves.toMatchSnapshot();
    });
  });

  describe('4 cols', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(demo4Col)).resolves.toMatchSnapshot();
    });
  });
});
