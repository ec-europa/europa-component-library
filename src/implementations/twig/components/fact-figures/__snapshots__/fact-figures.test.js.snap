// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fact and figures 1 col renders correctly 1`] = `
<jest>
  <div
    class="ecl-fact-figures ecl-fact-figures--col-1"
  >
    <div
      class="ecl-fact-figures__items"
    >
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#infographic"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc eu est nec orci rhoncus venenatis. Donec rutrum lacus eget lobortis eleifend. Duis cursus sapien ac felis bibendum, in sagittis turpis porttitor. Aliquam sed mauris et orci efficitur mollis. Donec eu dolor in turpis sollicitudin malesuada.
        </div>
      </div>
    </div>
    <div
      class="ecl-fact-figures__view-all"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-link--icon ecl-fact-figures__view-all-link"
        href="/example"
      >
        <span
          class="ecl-link__label"
        >
          View all
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--xs ecl-icon--flip-horizontal ecl-link__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#arrow-left"
          />
        </svg>
      </a>
    </div>
  </div>
</jest>
`;

exports[`Fact and figures 1 col renders correctly when centerd 1`] = `
<jest>
  <div
    class="ecl-fact-figures ecl-fact-figures--col-1 ecl-fact-figures--centered"
  >
    <div
      class="ecl-fact-figures__items"
    >
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#infographic"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc eu est nec orci rhoncus venenatis. Donec rutrum lacus eget lobortis eleifend. Duis cursus sapien ac felis bibendum, in sagittis turpis porttitor. Aliquam sed mauris et orci efficitur mollis. Donec eu dolor in turpis sollicitudin malesuada.
        </div>
      </div>
    </div>
    <div
      class="ecl-fact-figures__view-all"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-link--icon ecl-fact-figures__view-all-link"
        href="/example"
      >
        <span
          class="ecl-link__label"
        >
          View all
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--xs ecl-icon--flip-horizontal ecl-link__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#arrow-left"
          />
        </svg>
      </a>
    </div>
  </div>
</jest>
`;

exports[`Fact and figures 1 col renders correctly whit large icons 1`] = `
<jest>
  <div
    class="ecl-fact-figures ecl-fact-figures--col-1"
  >
    <div
      class="ecl-fact-figures__items"
    >
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--2xl ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#infographic"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc eu est nec orci rhoncus venenatis. Donec rutrum lacus eget lobortis eleifend. Duis cursus sapien ac felis bibendum, in sagittis turpis porttitor. Aliquam sed mauris et orci efficitur mollis. Donec eu dolor in turpis sollicitudin malesuada.
        </div>
      </div>
    </div>
    <div
      class="ecl-fact-figures__view-all"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-link--icon ecl-fact-figures__view-all-link"
        href="/example"
      >
        <span
          class="ecl-link__label"
        >
          View all
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--xs ecl-icon--flip-horizontal ecl-link__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#arrow-left"
          />
        </svg>
      </a>
    </div>
  </div>
</jest>
`;

exports[`Fact and figures 1 col renders correctly with extra attributes 1`] = `
<jest>
  <div
    class="ecl-fact-figures ecl-fact-figures--col-1"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <div
      class="ecl-fact-figures__items"
    >
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#infographic"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc eu est nec orci rhoncus venenatis. Donec rutrum lacus eget lobortis eleifend. Duis cursus sapien ac felis bibendum, in sagittis turpis porttitor. Aliquam sed mauris et orci efficitur mollis. Donec eu dolor in turpis sollicitudin malesuada.
        </div>
      </div>
    </div>
    <div
      class="ecl-fact-figures__view-all"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-link--icon ecl-fact-figures__view-all-link"
        href="/example"
      >
        <span
          class="ecl-link__label"
        >
          View all
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--xs ecl-icon--flip-horizontal ecl-link__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#arrow-left"
          />
        </svg>
      </a>
    </div>
  </div>
</jest>
`;

exports[`Fact and figures 1 col renders correctly with extra class names 1`] = `
<jest>
  <div
    class="ecl-fact-figures ecl-fact-figures--col-1 custom-class custom-class--test"
  >
    <div
      class="ecl-fact-figures__items"
    >
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#infographic"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc eu est nec orci rhoncus venenatis. Donec rutrum lacus eget lobortis eleifend. Duis cursus sapien ac felis bibendum, in sagittis turpis porttitor. Aliquam sed mauris et orci efficitur mollis. Donec eu dolor in turpis sollicitudin malesuada.
        </div>
      </div>
    </div>
    <div
      class="ecl-fact-figures__view-all"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-link--icon ecl-fact-figures__view-all-link"
        href="/example"
      >
        <span
          class="ecl-link__label"
        >
          View all
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--xs ecl-icon--flip-horizontal ecl-link__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#arrow-left"
          />
        </svg>
      </a>
    </div>
  </div>
</jest>
`;

exports[`Fact and figures 2 cols renders correctly 1`] = `
<jest>
  <div
    class="ecl-fact-figures ecl-fact-figures--col-2"
  >
    <div
      class="ecl-fact-figures__items"
    >
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#infographic"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc eu est nec orci rhoncus venenatis. Donec rutrum lacus eget lobortis eleifend. Duis cursus sapien ac felis bibendum, in sagittis turpis porttitor. Aliquam sed mauris et orci efficitur mollis. Donec eu dolor in turpis sollicitudin malesuada.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#spreadsheet"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Sed hendrerit
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Turpis varius congue venenatis, erat dui feugiat felis.
        </div>
      </div>
    </div>
    <div
      class="ecl-fact-figures__view-all"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-link--icon ecl-fact-figures__view-all-link"
        href="/example"
      >
        <span
          class="ecl-link__label"
        >
          View all
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--xs ecl-icon--flip-horizontal ecl-link__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#arrow-left"
          />
        </svg>
      </a>
    </div>
  </div>
</jest>
`;

exports[`Fact and figures 3 cols renders correctly 1`] = `
<jest>
  <div
    class="ecl-fact-figures ecl-fact-figures--col-3"
  >
    <div
      class="ecl-fact-figures__items"
    >
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#infographic"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc eu est nec orci rhoncus venenatis. Donec rutrum lacus eget lobortis eleifend. Duis cursus sapien ac felis bibendum, in sagittis turpis porttitor. Aliquam sed mauris et orci efficitur mollis. Donec eu dolor in turpis sollicitudin malesuada.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#spreadsheet"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Sed hendrerit
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Turpis varius congue venenatis, erat dui feugiat felis.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#growth"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Donec suscipit interdum augue, ac dapibus eros finibus a.
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Cras vestibulum efficitur mi, quis porta tellus rutrum ut. Quisque at pulvinar sem.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Aenean dapibus
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Aliquam lacinia diam eu sem malesuada, in interdum ante bibendum.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#regulation"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc condimentum sapien ut nibh finibus suscipit vitae at justo. Morbi quis odio faucibus, commodo tortor id, elementum libero.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#image"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Sed hendrerit
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Turpis varius congue venenatis, erat dui feugiat felis.
        </div>
      </div>
    </div>
    <div
      class="ecl-fact-figures__view-all"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-link--icon ecl-fact-figures__view-all-link"
        href="/example"
      >
        <span
          class="ecl-link__label"
        >
          View all
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--xs ecl-icon--flip-horizontal ecl-link__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#arrow-left"
          />
        </svg>
      </a>
    </div>
  </div>
</jest>
`;

exports[`Fact and figures 4 cols renders correctly 1`] = `
<jest>
  <div
    class="ecl-fact-figures ecl-fact-figures--col-4"
  >
    <div
      class="ecl-fact-figures__items"
    >
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#infographic"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc eu est nec orci rhoncus venenatis. Donec rutrum lacus eget lobortis eleifend. Duis cursus sapien ac felis bibendum, in sagittis turpis porttitor. Aliquam sed mauris et orci efficitur mollis. Donec eu dolor in turpis sollicitudin malesuada.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#spreadsheet"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Sed hendrerit
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Turpis varius congue venenatis, erat dui feugiat felis.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#growth"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Donec suscipit interdum augue, ac dapibus eros finibus a.
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Cras vestibulum efficitur mi, quis porta tellus rutrum ut. Quisque at pulvinar sem.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Aenean dapibus
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Aliquam lacinia diam eu sem malesuada, in interdum ante bibendum.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#regulation"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc condimentum sapien ut nibh finibus suscipit vitae at justo. Morbi quis odio faucibus, commodo tortor id, elementum libero.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#image"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Sed hendrerit
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Turpis varius congue venenatis, erat dui feugiat felis.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#global"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Donec suscipit interdum augue, ac dapibus eros finibus a.
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Cras vestibulum efficitur mi, quis porta tellus rutrum ut. Quisque at pulvinar sem.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--l ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#presentation"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Aenean dapibus
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Aliquam lacinia diam eu sem malesuada, in interdum ante bibendum.
        </div>
      </div>
    </div>
    <div
      class="ecl-fact-figures__view-all"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-link--icon ecl-fact-figures__view-all-link"
        href="/example"
      >
        <span
          class="ecl-link__label"
        >
          View all
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--xs ecl-icon--flip-horizontal ecl-link__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#arrow-left"
          />
        </svg>
      </a>
    </div>
  </div>
</jest>
`;
